### PR TITLE
TM-798 add custom variables API

### PR DESCRIPTION
**Pull request by ArekZc**
TM-812 As a user I want to use Random Number custom variable

### CODE REVIEW

File: app/AppKernel.php [hunk 1]
```diff
@@ -53,12 +53,15 @@ public function registerBundles()
...
             new SevenTag\Api\TriggerBundle\SevenTagTriggerBundle(),
             new SevenTag\Api\AppBundle\SevenTagAppBundle(),
+            new SevenTag\Api\VariableBundle\SevenTagVariableBundle(),
 
             new SevenTag\Plugin\GoogleAnalyticsCustomTemplateBundle\SevenTagPluginGoogleAnalyticsCustomTemplateBundle(),
             new SevenTag\Plugin\GoogleAdwordsCustomTemplateBundle\SevenTagPluginGoogleAdwordsCustomTemplateBundle(),
             new SevenTag\Plugin\PiwikCustomTemplateBundle\SevenTagPluginPiwikCustomTemplateBundle(),
             new SevenTag\Plugin\CrazyEggCustomTemplateBundle\SevenTagPluginCrazyEggCustomTemplateBundle(),
             new SevenTag\Plugin\FacebookRetargetingPixelCustomTemplateBundle\SevenTagPluginFacebookRetargetingPixelCustomTemplateBundle()
+
```
Comment (kallosz) [inline app/AppKernel.php]: ?? For what is this?

File: app/DoctrineMigrations/Version20151009152220.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        $this->addSql('ALTER TABLE variable ADD CONSTRAINT FK_CC4D878DBC21F742 FOREIGN KEY (container_id) REFERENCES container (id) ON DELETE CASCADE');
+        $this->addSql('ALTER TABLE variable ADD CONSTRAINT FK_CC4D878D4BBC2705 FOREIGN KEY (version_id) REFERENCES version (id) ON DELETE CASCADE');
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        // this down() migration is auto-generated, please modify it to your needs
```
Comment (kallosz) [inline app/DoctrineMigrations/Version20151009152220.php]: not needed comment

File: app/DoctrineMigrations/Version20151009152220.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * Auto-generated Migration: Please modify to your needs!
+ */
+class Version20151009152220 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
```
Comment (kallosz) [inline app/DoctrineMigrations/Version20151009152220.php]: ?? rly? remove this please ;-)

File: app/DoctrineMigrations/Version20151009152220.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (kallosz) [inline app/DoctrineMigrations/Version20151009152220.php]: Put info about migration.

File: composer.json [hunk 5]
```diff
@@ -106,3 +106,4 @@
         }
     }
 }
+
```
Comment (kallosz) [inline composer.json]: this changes crush synchronization between composer.json and composer.lock
image

File: src/SevenTag/Api/TriggerBundle/Comperator/Comparator.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (kallosz) [inline src/SevenTag/Api/TriggerBundle/Comperator/Comparator.php]: this file has syntax error. also folder name `Comperator` ?

File: src/SevenTag/Api/VariableBundle/Command/VariableTypesCommand.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
+<?php
```
Comment (kallosz) [inline src/SevenTag/Api/VariableBundle/Command/VariableTypesCommand.php]: this should be migration script not command.

File: src/SevenTag/Api/VariableBundle/Controller/VariableTypeRestController.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+     *  statusCode=200,
+     *  serializerEnableMaxDepthChecks="true",
+     *  serializerGroups={"variableType"}
+     * )
+     *
+     * @return array
+     */
+    public function getVariablesTypesAction()
+    {
+
```
Comment (kallosz) [inline src/SevenTag/Api/VariableBundle/Controller/VariableTypeRestController.php]: what are the new lines?

File: src/SevenTag/Api/VariableBundle/Registry/VariableTypeRegistry.php [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+use SevenTag\Component\Variable\Model\VariableType;
+use SevenTag\Component\Variable\Model\VariableTypeInterface;
+
+/**
+ * Class VariableTypeRegistry
+ * @package SevenTag\VariableBundle\Registry
+ */
+class VariableTypeRegistry
+{
+
```
Comment (kallosz) [inline src/SevenTag/Api/VariableBundle/Registry/VariableTypeRegistry.php]: what are the new lines?

File: src/SevenTag/Api/TriggerBundle/Comperator/Comparator.php (hunk 6 above)
Comment (ArekZc) [inline src/SevenTag/Api/TriggerBundle/Comperator/Comparator.php]: Need to remove this file and folder

File: composer.json (hunk 5 above)
Comment (ArekZc) [inline composer.json]: "doctrine/migrations": " 1.0.*@dev" this crush synchronization

**Pull request merged**